### DISCUSSION TITLE
2FA su HomeAssistant raspberry PI

### DISCUSSION THREAD

**Discussion by Angelodomotica** [indomus/forum#3294] [GENERALE - Discussioni generiche]
Buongiorno a tutti,
premesso che sono un neofita alle prime armi con HA intsllato su un raspberry, volevo chiedere come posso disattivare autenticazione a due fattori da console? avendolo configurare con google authenticator ma ho perso il telefono e adesso con il nuovo cell. google authenticator non memorizza gli accesi com authy. qualcuno potrebbe incollarmi i  comandi da eseguire da console grazie? 
p.s. ho googlato tuti dicono ho risolto ma nessuno scrive che comando bisogna dare e non ho mai installato il servizio ssh oppure potrei installare ssh da console e modificare questo file .yaml così che non m,i chieda piu il 2fa

**Comment by snakuzzo** [comment]
Non so se è documentato ufficialmente, in ogni caso da quello che leggo non mi pare di vedere nulla di così complicato.
Innanzi tutto, che tipo di installazione hai? HA OS oppure Core ?

I passi di fatto sarebbero i seguenti:

- Vai sotto la dir della configurazione di HA
- Poi sotto `.storage` e rinomini il file `auth_module.totp` in `auth_module.totp.tmp`
- Riavvii home assistant


Se hai HA OS...per la CLI o colleghi un monitor e una tastiera oppure installi ad esempio l'addon Advanced SSH & Web Terminal

![image](https://github.com/user-attachments/assets/55ab8f83-f511-468a-8740-7f0aaeecd0db)

e riavvii Core

Se invece hai Core in venv o docker, avrai sicuramente accesso alla shell (in locale o in SSH)...stessi passi di cui sopra

es.

```
cd /home/pi/homeassistant/.storage
mv auth_module.totp auth_module.totp.tmp
```

e riavvii Core

**Comment by Angelodomotica** [comment]
@snakuzzo grazie della pronta risposta ho installato il HA core come da foto allegata ma tramite ssh con putty non mi fa entrare che porta devo usare la 22 giusto? googlando ho dato questi comandi ma non sapevo poi come arrivare al file.storage 
login
docker exec -it homeassistant bash
eventualemte da terminale posso installare ssh e web terminal? 
dimenticavo ho collegato monitor tastiera e mouse
![HA Version](https://github.com/user-attachments/assets/19df7d4f-2aec-4510-93e6-5fc836f0f8ea)

**Reply by snakuzzo** [reply]
non hai installato core, quella è la CLI di HA che è presente solo su HA OS e HA Supervised.
Leggi cosa ti ho scritto io

**Comment by Angelodomotica** [comment]
ho riletto il tutto e da quello che hai scritto dovrei entrare e dare questi comandi 
cd config/.storage
e poi
storage mv auth_module.totp auth_module.totp.tmp

per riavviare da console è giusto questo comando?
sudo systemctl restart home-assistant@homeassistant

invece che comando devo usare su HA OS per installare ssh e web terminal da console?

cmq oggi proverò e ti aggiornerò @snakuzzo grazie per il momento

**Reply by snakuzzo** [reply]
stai confondendo fischi con fiaschi.
Non puoi dare i comandi che hai indicato su HA OS.
Una volta fatta la modifica, puoi semplicemente riavviare Core da GUI

![image](https://github.com/user-attachments/assets/e0681b55-c2ab-4e52-9c9e-b4da69d2e0e6)

**Comment by Angelodomotica** [comment]
il problema che dopo aver fatto la modifica se non riavvio si applica? visto che non entro non avendo il 2fa?
quindi non posso installare ssh e web terminal su HA OS?

questi per fare la modifica sono giusti vero? 

cd config/.storage
e poi
storage mv auth_module.totp auth_module.totp.tmp

**Comment by Angelodomotica** [comment]
@snakuzzo ho digitato tutti i comandi ma mi da sempre errore 

cd config/.storage

sempre errore 

come posso risolvere?

**Comment by Angelodomotica** [comment]
guarda le foto 
![1](https://github.com/user-attachments/assets/f72b19d0-53c7-48be-9989-90bb1383d6df)
![Uploading 2.jpeg…]()

**Reply by snakuzzo** [reply]
hai ragione. Non hai la bash se ti colleghi con mouse e tastiera.
Quindi devi fare così...

![image](https://github.com/user-attachments/assets/078b9b70-2d98-4827-b23b-c7adba7d9bdb)

(occhio che il layout della tastiera è americano)

**Comment by Angelodomotica** [comment]
@snakuzzo Buongiorno, ha funzionato così grazie quindi il mio è HA OS supervised. Me ne sono accorto che la tastiera è americana -:).
posso farti un altra domanda oppure se hai un canale privato per non sporcare troppo il post.

vorrei attivare cloudfare per accesso da esterno e iniziare a smanettare un pò e sicuramente chiedere tante altre cose essendo un neofita

https://indomus.it/guide/collegarsi-da-remoto-a-home-assistant-via-reti-provider-tradizionali-o-che-usino-nat-via-cloudflare/#coredocker

ho letto questa guida poi la provo ma non c'è scrito sul mio dominio che ho creato esempio casamia.com che indirizzo vado a mettere per puntare? non c'è scritto nella guida anxche perchè il 90% degli operatori non da ip statico

**Reply by snakuzzo** [reply]
1. Se fai questa osservazione sulla tua installazione stiamo messi male. Dovresti sapere che tipo di installazione hai. O hai HA OS o HA Supervised, sono due cose diverse. Leggi [QUI](https://indomus.it/guide/home-assistant-hassio-o-home-assistant-core-quale-installazione-fa-al-caso-mio/)
2. Niente messaggi o canali privati. Se hai bisogno di altre cose, cerca prima sul forum e poi, nel caso, chiedi (sono le regole base per chi frequenta una community). Da quello che leggo, non hai bisogno di aprire nuovi topic, visto che è un tema discusso parecchie volte qui, in chat e anche nei commenti sul sito.
3. Alla tua ultima domanda io cosa dovrei rispondere ? Non si capisce una fava di quello che hai chiesto e non c'entra nulla il discorso IP statico o dinamico, che tra l'altro TU stai confondendo con IP pubblico o privato. Gran parte degli operatori danno IP privato, non statico.

Detto ciò. questa discussione può anche essere chiusa

**Comment by Angelodomotica** [comment]
1. Ho installato HA supervised l'ho scritto 
2. proverò a cercare sul forum per quanto riguarda cloudfare
3. penso di essere stato chiaro il dominio per funzionare ha bisogno di un ip dove puntare come un dns come duck gli operatori danno ip pubblico intendevop ch ecambia e non posso puntare ad un dominio un ip che cambia non so se sono stato chiaro

**Reply by snakuzzo** [reply]
sul punto 3 permettimi di dire che non hai capito di che IP si parla. Lì si parla di ip del DNS Serer non dell'IP tuo.
Inoltre, stai facendo tutto un po' a caso...perché sei arrivato su una guida senza leggere quella che viene prima (e che tra le altre cose è linkata proprio nell'articolo che stavi leggendo) -> https://indomus.it/guide/configurare-cloudflare-per-il-proprio-dominio-personale-per-protezione-reverse-proxy-cdn-e-altro/

Ora, per favore, riparti dalla discussione che ti ho linkato ora e... discussione chiusa
